File: core/java/lib/src/main/java/com/microsoft/kiota/core/serialization/JsonSerializationWriterFactory.java
package com.microsoft.kiota.core.serialization;

import java.util.Objects;

import javax.annotation.Nonnull;

import com.microsoft.kiota.serialization.SerializationWriter;
import com.microsoft.kiota.serialization.SerializationWriterFactory;

public class JsonSerializationWriterFactory implements SerializationWriterFactory {
    private static final String validContentType = "application/json";
    @Override
    @Nonnull
    public SerializationWriter getSerializationWriter(@Nonnull final String contentType) {
        Objects.requireNonNull(contentType, "parameter contentType cannot be null");
        if(contentType.isEmpty()) {
            throw new NullPointerException("contentType cannot be empty");
        } else if (!contentType.equals(validContentType)) {
            throw new IllegalArgumentException("expected a " + validContentType + " content type");
        }
        return new JsonSerializationWriter();
    }
}

File: serialization/ruby/json/microsoft_kiota_serialization/lib/microsoft_kiota_serialization/json_serialization_writer_factory.rb
require 'microsoft_kiota_abstractions'

module MicrosoftKiotaSerialization
  class JsonSerializationWriterFactory
    include MicrosoftKiotaAbstractions::SerializationWriterFactory

    def get_valid_content_type
      'application/json'
    end

    def get_serialization_writer(content_type)
      if !content_type
        raise StandardError, 'content type cannot be undefined or empty'
      elsif get_valid_content_type != content_type
        raise StandardError, `expected a #{get_valid_content_type} content type`
      end

      JsonSerializationWriter.new
    end
  end
end
